templates/zerver/integrations/redmine.md

```diff
@@ -39,7 +39,7 @@ you want these settings to apply.
 To configure Zulip notifications for a particular Redmine project,
 visit the project's **Settings** page.
 
-In either case, fill out zulip server by `{{ external_api_path_subdomain }}`
+In either case, fill out zulip server by `{{ api_url_scheme_relative }}`
```

Comment:
This should probably be `api_url`?  Worth reading https://github.com/zulip/zulip-redmine-plugin/blob/master/redmine_zulip/lib/zulip_hooks.rb#L108 ...

Also, this sentence doesn't make any sense; worth fixing that too; maybe something like:

+In either case, fill out the settings with the Zulip server (`{{
+api_url }}`), the bot's email address and API key, and the Zulip
+stream that should receive notifications. Apply your changes.

Reply:
Huh, that code in the plugin looks kind of wacky:
```
   def zulip_api_basename()
     basename = Setting.plugin_redmine_zulip[:zulip_server].sub %r{^https?:(//|\\\\)}i, ''
     if not basename.end_with? "/api"
         basename = basename.concat "/api"
     end
     return basename
   end
# [...]
    def send_zulip_message(content, project)
        # [...]
        req = Net::HTTP::Post.new(zulip_api_basename() + "/v1/messages")
```
Looking around the web quickly, I think the issue is that this Ruby library's API very closely parallels the actual HTTP interaction -- so this object that's being constructed will be used on an already-established connection, so it can supply the URL's authority and path+query (to go in the Host header and the POST line respectively) but not the schema (which is already set by establishing the connection.

... Oh, and I suspect the reason for this schemaless advice is this line:
```
        http = Net::HTTP.new(zulip_server(), zulip_port())
```
where `zulip_server()` just returns the config value.

The [first tutorial I find](http://www.rubyinside.com/nethttp-cheat-sheet-2940.html) searching the web for this library has you start from a proper full URL, parse it, and use the fields from the parsed URL rather than try to string-manipulate. That makes a lot of sense to me. Alternatively what it actually suggests is to use some other library with a nicer API. :-)

... OK, I don't think this merits me spending more than a few minutes sorting through it right now. It looks likely that the plugin as it stands can break if the schema is included, so I'll Do No Harm in touching the docs. It'd be good for someone to clean up the plugin at some point, if it has any users.